lib/sigh/developer_center.rb

```diff
@@ -59,49 +59,53 @@ def maintain_app_certificate(force = nil)
+              details = profile_details(current_cert['provisioningProfileId'])
 
-              next if profile_name && details['provisioningProfile']['name'] != profile_name
+              if details['provisioningProfile']['appId']['identifier'] == bundle_id
 
-              # that's an Ad Hoc profile. I didn't find a better way to detect if it's one ... skipping it
-              next if @type == APPSTORE && details['provisioningProfile']['deviceCount'] > 0
+                next if profile_name && details['provisioningProfile']['name'] != profile_name
 
-              # that's an App Store profile ... skipping it
-              next if @type != APPSTORE && details['provisioningProfile']['deviceCount'] == 0
+                # that's an Ad Hoc profile. I didn't find a better way to detect if it's one ... skipping it
+                next if @type == APPSTORE && details['provisioningProfile']['deviceCount'] > 0
 
-              # We found the correct certificate
-              if force && @type != DEVELOPMENT
-                provisioningProfileId = current_cert['provisioningProfileId']
-                renew_profile(provisioningProfileId) # This one needs to be forcefully renewed
-                return maintain_app_certificate(false) # recursive
-              elsif current_cert['status'] == 'Active'
-                return download_profile(details['provisioningProfile']['provisioningProfileId']) # this one is already finished. Just download it.
-              elsif ['Expired', 'Invalid'].include? current_cert['status']
-                renew_profile(current_cert['provisioningProfileId']) # This one needs to be renewed
-                return maintain_app_certificate(false) # recursive
-              end
+                # that's an App Store profile ... skipping it
+                next if @type != APPSTORE && details['provisioningProfile']['deviceCount'] == 0
+
+                # We found the correct certificate
+                if force && @type != DEVELOPMENT
+                  provisioningProfileId = current_cert['provisioningProfileId']
+                  renew_profile(provisioningProfileId) # This one needs to be forcefully renewed
+                  return maintain_app_certificate(false) # recursive
+                elsif current_cert['status'] == 'Active'
+                  return download_profile(details['provisioningProfile']['provisioningProfileId']) # this one is already finished. Just download it.
+                elsif ['Expired', 'Invalid'].include? current_cert['status']
+                  renew_profile(current_cert['provisioningProfileId']) # This one needs to be renewed
+                  return maintain_app_certificate(false) # recursive
+                end
 
-              break
+                break
+              end
             end
-          end
 
-          if page_size == profile_count
-            page_index += 1
-          else
-            has_all_profiles = true
+            if page_size <= profile_count
```

Comment:
Potentially, but only if there was a bug in Apples API that would return more than the `page_size`, more defensive than thinking it'd actually get hit.